### PR TITLE
feat(nuxt): support for extending error.vue in layers

### PR DESCRIPTION
**Pull request by manniL**
### 🔗 Linked issue

Resolves https://github.com/nuxt/framework/issues/9520
Related: https://github.com/nuxt/framework/issues/3222

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Adds layer/`extends` support for `error.vue`

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt/src/core/app.ts [hunk 1]
```diff
@@ -74,7 +74,12 @@ export async function resolveApp (nuxt: Nuxt, app: NuxtApp) {
 
   // Resolve error component
   if (!app.errorComponent) {
-    app.errorComponent = (await findPath(['~/error'])) || resolve(nuxt.options.appDir, 'components/nuxt-error-page.vue')
+    app.errorComponent = await findPath(
+      [
+        ...nuxt.options._layers.map(layer => `${layer.config.srcDir}/error`),
```
Comment (pi0) [inline packages/nuxt/src/core/app.ts:79]: Would be better to use pathe.join here

File: packages/nuxt/src/core/app.ts [hunk 2]
```diff
@@ -74,7 +74,12 @@ export async function resolveApp (nuxt: Nuxt, app: NuxtApp) {
...
+        resolve(nuxt.options.appDir, 'components/nuxt-error-page.vue')
```
Comment (pi0) [inline packages/nuxt/src/core/app.ts:80]: Can we please keep fallback outside of search path like before?

Comment (manniL) [reply packages/nuxt/src/core/app.ts]: Sure!

**Comment by manniL** [comment]
I'll refactor the rest of the file in another PR to make use of `pathe.join` + keeping fallback out of `findPath` - for e.g. `App.vue`

File: packages/nuxt/src/core/app.ts [hunk 3]
```diff
@@ -74,7 +74,9 @@ export async function resolveApp (nuxt: Nuxt, app: NuxtApp) {
 
   // Resolve error component
   if (!app.errorComponent) {
-    app.errorComponent = (await findPath(['~/error'])) || resolve(nuxt.options.appDir, 'components/nuxt-error-page.vue')
+    app.errorComponent = await findPath(
+      nuxt.options._layers.map(layer => join(layer.config.srcDir, 'error'))
+    ) ?? resolve(nuxt.options.appDir, 'components/nuxt-error-page.vue')
```
Comment (danielroe) [inline packages/nuxt/src/core/app.ts:79]:
```suggestion
    app.errorComponent = (await findPath(
      nuxt.options._layers.map(layer => join(layer.config.srcDir, 'error'))
    )) ?? resolve(nuxt.options.appDir, 'components/nuxt-error-page.vue')
```

Comment (manniL) [reply packages/nuxt/src/core/app.ts]: Good catch ☺️

**Review by Atinux — APPROVED** [review]

**Pull request merged**